server/api.go

```diff
@@ -503,6 +501,13 @@ func (a *API) oauthRedirectHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	err = a.p.setNotificationPreference(mmUserID, true)
+	if err != nil {
+		a.p.API.LogWarn("Failed to update user preferences on connect", "user_id", mmUserID, "error", err.Error())
+		http.Error(w, "error updating the preferences", http.StatusInternalServerError)
```

Comment:
should there be a reference to teams in the error logs/messages? Same way as you have when it is successful. Or are admins/users able to tell this is related to the plugin?

Reply:
Ooh, good question! In this case, we get the benefit of the server automatically tagging all logs with the `plugin_id`, e.g.:

```json
{"timestamp":"2024-09-24 15:10:08.987 Z","level":"warn","msg":"Failed to update user preferences on connect","plugin_id":"com.mattermost.msteams-sync","user_id":"rcnuja8z8p8mb811p989kpaosh"}
```